lib/dgram.js

```diff
@@ -138,7 +138,6 @@ function Socket(type, listener) {
   this._bindState = BIND_STATE_UNBOUND;
   this[async_id_symbol] = this._handle.getAsyncId();
   this.type = type;
-  this.fd = null; // compatibility hack
```

Comment:
hmm... @nodejs/tsc ... do we know if this could break anyone? It may be minor, but we should likely do a deprecation on this before just removing it.

Reply:
I think a simple semver-major removal is fine for this, given that it seems pretty unusable.